### PR TITLE
Start of bank helpers for ComponentInfo

### LINKED ISSUE #23703: Helper Methods for Checking Banks and Detectors in ComponentInfo
This is related to a deal with duplication of checks for whether or not a detector is within a Grid, Rectangular or Structured bank. See `MoveInstrumentComponent` as an example.

### PR DESCRIPTION
**Pull request by LamarMoore**
**Description of work.**

Added some bank helpers for the ComponentInfo. First helper to check whether or not detectors are fixed within a bank.

**To test:**

Code review

Fixes #23703. 

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/IndividualTicketTesting/)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

File: Framework/Geometry/src/Instrument/ComponentInfoBankHelpers.cpp [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+
+namespace Mantid {
+namespace Geometry {
+namespace ComponentInfoBankHelpers {
+bool isDetectorFixedInBank(const ComponentInfo &compInfo,
+                           const size_t detIndex) {
+  auto parent = compInfo.parent(detIndex);
+  auto grandParent = compInfo.parent(parent);
+  auto grandParentType = compInfo.componentType(grandParent);
+
```
Comment (OwenArnold) [inline Framework/Geometry/src/Instrument/ComponentInfoBankHelpers.cpp]: You could also test that `parent` and `grandParent` are unique indexes and are distinct from `detIndex`. If your detector was attached directly to the instrument, there would be no grandparent, and the parent and grandparent indexes would be the same. So could just return false at that point. Maybe this isn't worth doing though?

File: Framework/Geometry/test/ComponentInfoBankHelpersTest.h [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+
+  void test_DetectorFixedInBankTrueForRectangularBank() {
+    auto rectInstr =
+        ComponentCreationHelper::createTestInstrumentRectangular(1, 2);
+    auto wrappers = InstrumentVisitor::makeWrappers(*rectInstr);
+    const auto &componentInfo = *wrappers.first;
+
+    TS_ASSERT(isDetectorFixedInBank(componentInfo, 0));
+  }
+
```
Comment (OwenArnold) [inline Framework/Geometry/test/ComponentInfoBankHelpersTest.h]: Possibly missing a test here for something that isn't a detector at all. Additionally, what about a monitor attached to an instrument.

**Review by OwenArnold — CHANGES_REQUESTED** [review]
Why didn't you update `Instrument::makeLegacyParameterMap`? https://github.com/mantidproject/mantid/blob/master/Framework/Geometry/src/Instrument.cpp#L1316

Overall solution looks good.

File: Framework/Geometry/src/Instrument/ComponentInfoBankHelpers.cpp (hunk 1 above)
Comment (LamarMoore) [reply Framework/Geometry/src/Instrument/ComponentInfoBankHelpers.cpp]: The test would fail as long as the grandparent type isn't match regardless.

**Review by OwenArnold — APPROVED** [review]

**Pull request merged**